### ISSUE TITLE
Add intangible asset register with amortization

### ISSUE THREAD

**Issue by bbengt1** [bbengt1/3d-print-sales#252]
## User problem
Annual CAD/slicer subscriptions, bulk STL/asset-pack purchases, domain/brand acquisitions, multi-year listing fees deliver value over time but currently expense in full the month they're purchased. We need a symmetric counterpart to the fixed-asset register (#238) for intangibles.

## Scope
**In scope (v1):** structural mirror of #238 against intangibles.
- New `IntangibleAsset` and `AmortizationEntry` models with fields paralleling `FixedAsset` and `DepreciationEntry`. Differences: no physical-equipment link (no `printer_id` analog), `useful_life_months` typically shorter, no salvage value (default 0).
- Same two methods: straight-line and declining-balance.
- Same manual posting model: "Post amortization through period end".
- Same disposal flow: write-off (loss = remaining book value) or sale-with-proceeds (gain/loss).
- COA seeding: *Intangible Assets* (asset), *Accumulated Amortization* (contra-asset), *Amortization Expense* (expense), *Loss on Disposal of Intangibles* (expense), *Gain on Disposal of Intangibles* (income).
- Allocator scope `intangible_asset` format `IA-{year}-{value:04d}`.
- Frontend: `/finance/intangible-assets` list + detail/edit + disposal modal — same shape as the fixed-asset page.

**Out of scope:** component intangibles, mid-month proration beyond fixed-asset's behavior, tax-method amortization separate from book.

## Acceptance criteria
- [ ] Migration adds `intangible_assets` and `amortization_entries`; downgrade reverses.
- [ ] COA seeding adds the five accounts; existing businesses get idempotent seeder.
- [ ] Math parity with #238: SL across N months, DDB with SL crossover, idempotent reposting through a period.
- [ ] Disposal flow with proceeds (gain/loss) and without (full write-off as loss).
- [ ] Period-lock guard on amortization posting and disposal.
- [ ] Edit lock once entries exist (cost/method/life immutable).
- [ ] Backend tests mirror the fixed-asset suite.
- [ ] Frontend list, create, edit, post, dispose flows work end-to-end.
- [ ] New doc `docs/intangible_assets.md` (cross-references #238 for shared concepts).
- [ ] Gap doc updated.

## Validation
- `python3 -m pytest backend/tests -q`
- `cd frontend && npm run build`

## Source
Manager.io gap walkthrough, Tier 3 #13. Symmetric mirror of #238.
